File: Array, Q, Simulate/Array, Q, Simulation/Longest Common Prefix.java
class Solution {
    public String longestCommonPrefix(String[] strs) {
        if (strs == null || strs.length == 0) 
        {
            return "";
        }
        
        //Find the shortest word
        String shortest = strs[0];
        for (String word : strs) 
        {
            if (word.length() < shortest.length())
             {
                shortest = word;
            }
        }
        
        //Check characters one by one
        for (int i = 0; i < shortest.length(); i++) 
        {
            char c = shortest.charAt(i);
            
            
            for (String word : strs) 
            {
                if (word.charAt(i) != c) 
                {
                    //Return the prefix up to this point if mismatch
                    return shortest.substring(0, i);
                }
            }
        }
        
        
        return shortest;
    }
}

File: Array, Q, Simulate/Array, Q, Simulation/Longest Common Prefix.py
from typing import List

class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        if not strs:
            return ""
            
        # Find the shortest word 
        shortest = min(strs, key=len)
        
        for i, char in enumerate(shortest):
            
            for word in strs:
                if word[i] != char:
                    
                    return shortest[:i]
                    
        
        return shortest